src/goto-programs/goto_convert.cpp

```diff
@@ -857,25 +857,40 @@ void goto_convertt::convert_assume(
   dest.add(goto_programt::make_assumption(op, code.source_location()));
 }
 
-void goto_convertt::convert_loop_invariant(
+void goto_convertt::convert_loop_invariant_and_decreases(
```

Comment:
I'd just call it `convert_loop_contracts` now. Assigns clause stuff would also be added in this function eventually